### PR TITLE
Backport #12140 to chef-17

### PR DESCRIPTION
**Pull request by jasonwbarnett**
Backport #12140 to chef-17

### CODE REVIEW

**Review by johnmccrae — APPROVED** [review]
This all looks good, I want to do an Ad/Hoc build before merging this.

**Comment by johnmccrae** [comment]
@jasonwbarnett  can you do one more rebase and push it again? I corrected the problem with the Mac builds. I'll merge it after that.

**Comment by jasonwbarnett** [comment]
@johnmccrae The last commit on the `chef-17` branch was nine days ago. There is nothing to rebase. In other words, this branch is fully rebased off of `chef-17`.

**Pull request merged**
